index.rst

```diff
@@ -42,20 +42,156 @@
 
 .. sectnum::
 
-.. TODO: Delete the note below before merging new content to the master branch.
+.. _introduction:
+
+Introduction
+============
+
+At the summit it is sometimes desirable for instrument data to be analyzed rapidly after data acquisition in order to get quick feedback of data quality.
+The different use cases for processing data are described in :cite:`DMTN-111` and in this document we focus on data processing directly related to data acquisition.
+
+Data processing must be triggered in some way with enough information to be able to work out which data to process and which analysis pipeline to use.
+The unique identifier for every observation is the "Observation Identifier" (aka "obsid" aka "image name") that is guaranteed by the observatory systems to be unique for every observation regardless of telescope (main telescope vs AuxTel), instrument, or which controller was involved to take the data.
+This obsid is issued by the camera control system whenever data are acquired.
+Additionally, there is a "grouping" concept that can be used by scripts to indicate that observations taken with the same group identifier are related [*]_.
+The group ID is how standard visits will be observed with both snaps being part of the same group.
+The group ID will be provided to each script by the ScriptQueue and will be unique for that execution of the script.
+
+Since DM software always mediates data access via a Data Butler this leads to a requirement that raw data must always be retrievable from a butler with a Data ID solely containing the obsid, and optionally, the detector name or group name.
+:cite:`DMTN-111` also states that some event must be issued whenever new data have been ingested and that that event must contain the obsid.
+
+.. [*] In :cite:`LTS-836` the group ID is called a "transaction ID" since it is associated with a single queue transaction.
+
+Processing Data as it Arrives
+=============================
+
+All triggered processing should be done through the "Gen3" middleware pipeline system consisting of a pipeline of``PipelineTask``.
+Each trigger will need to specify at minimum the name of the Pipeline to execute and the Data ID that should be processed.
```

Comment:
At this point the dataId will be the only information available to the OCPS. It could presumably query the butler to get the metadata associated with that dataId in order to get ObservationInfo parameters but a lot of those would already be in the registry). You might need to query the registry for the groupId in order to generate the correct graph for the combination pipeline or in theory the thing sending round the obsid indicating a file has been ingested could also send round the groupid.